client/landing/stepper/hooks/use-site-title.ts

```diff
@@ -0,0 +1,35 @@
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+import { isAnchorPodcastIdValid } from 'calypso/landing/stepper/hooks/use-is-anchor-fm';
+import { ONBOARD_STORE } from 'calypso/landing/stepper/stores';
+import usePodcastTitle from './use-podcast-title';
+
+export default function useSiteTitle(): void {
+	const { setSiteTitle } = useDispatch( ONBOARD_STORE );
+	const podcastTitle = usePodcastTitle();
+	const { siteTitle } = useSelect( ( select ) => select( ONBOARD_STORE ).getState() );
+	const hasSiteTitle = siteTitle.length > 0;
+
+	// When first loading a url including an ?anchor_podcast query param, clear
+	// the title.
+	// This allows the title retrieved from from backend to be applied, since
+	// we only apply it over blank titles to prevent wiping out custom titles
+	// set by the user.
+	// This also retains the ability to keep custom titles when reloading the
+	// browser during steps
+	const { search } = useLocation();
+	useEffect( () => {
+		const queryAnchorPodcast = new URLSearchParams( search ).get( 'anchor_podcast' );
+		if ( isAnchorPodcastIdValid( queryAnchorPodcast ) ) {
+			setSiteTitle( '' );
+		}
+	}, [] ); // Only run on initial mount
+
+	useEffect( () => {
+		if ( podcastTitle && ! hasSiteTitle ) {
```

Comment:
I think something in the logic here might be wonky, because using the back button into the podcast title step seems to re-fetch from the api.

Steps to reproduce: 
1) Open `/setup/podcastTitle?anchor_podcast={podcastid}`
2) Change the prepopulated name to something new
3) Submit and go to the next step
4) Click the back button
5) The title is populated with the Anchor API title, not the updated one from step 2.

Probably this is because the hook views this as a re-mount of the component.

Reply:
Yeah, I noticed this, too. It's also happening in the `/new` flow, so I may work on a fix in a future issue unless it's a blocker.

Reply:
Fair enough. Not a blocker for merge I guess but I think we wanna make sure the back-and-forth behavior between steps is clean before any launch.